### DISCUSSION TITLE
Is the `suspense` option all or nothing?

### DISCUSSION THREAD

**Discussion by jarrodpayne** [urql-graphql/urql#3411] [Q&A] [answered]
It appears the `suspense` option only works by providing a boolean to `createClient`. Does this mean there is no way to opt in and out of `suspense` mode per query? The `useQuery` api seems to accept `{  context: {  suspense: false } }`, but this seems to break my application and infinitely re-render.

**Comment by JoviDeCroock** [comment, marked as answer]
Context has to be memoized so useMemo or define a { suspense: false } context outside of the component

**Reply by jarrodpayne** [reply]
🤦🏻 Can't believe that one got me. Thanks!
